From 91f3d3d2be38803e9b472ae953f5a7051f2b6a3e Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Wed, 20 Mar 2024 05:18:59 +0200
Subject: [PATCH] [SW-165981] emb cmn: added strstr function

when emb common mods integrated for both baremetal and
preboot rtos, strstr function is required in one of the fit
function. So added strstr function.

Change-Id: I20d511e7f92a9378764bcb7a39926339a7ff4ab1
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/include/string.h b/include/string.h
index b74e7ae..ae5b7b2 100644
--- a/include/string.h
+++ b/include/string.h
@@ -27,5 +27,6 @@
 size_t strnlen(const char * s, size_t count);
 
 char * strrchr(const char * s, int c);
+char *strstr(const char *str, const char *substr);
 
 #endif /* _PREBOOT_STRING_H_ */
diff --git a/lib/common/string.c b/lib/common/string.c
index a315bf0..8a71fee 100644
--- a/lib/common/string.c
+++ b/lib/common/string.c
@@ -242,3 +242,28 @@
 			i++;
 	}
 }
+
+/**
+ * strstr - function finds the first occurrence of string2 in string1
+ * @str: The string to be searched
+ * @substr: The string to search for
+ */
+char *strstr(const char *str, const char *substr)
+{
+	while (*str) {
+		const char *s1 = str;
+		const char *s2 = substr;
+
+		while (*s1 && *s2 && *s1 == *s2) {
+			s1++;
+			s2++;
+		}
+
+		if (*s2 == '\0')
+			return (char *)str;
+
+		str++;
+	}
+
+	return NULL;
+}
